util/econftool.c

```diff
@@ -0,0 +1,654 @@
+/*
+  Copyright (C) 2019, 2020 SUSE LLC
+  Author: Dominik Gedon <dgedon@suse.de>
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+  The above copyright notice and this permission notice shall be included in all
+  copies or substantial portions of the Software.
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+  SOFTWARE.
+*/
+
+#include <errno.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <limits.h>
+
+#include "libeconf.h"
+
+static const char *utilname = "econftool";
+static const char *TMPPATH= "/tmp";
+static const char *TMPFILE_ORIG = "econftool.tmp";
+static const char *TMPFILE_EDIT = "econftool_edits.tmp";
+static const char *DROPINFILENAME = "90_econftool.conf";
+static bool isRoot = false;
+static bool isDropinFile = true;
+static econf_file *key_file = NULL;
+static econf_file *key_file_edit = NULL;
+
+static void deleteTmpFiles(void);
+static void diffGroups(char **, char **, char ***, char ***, char ***, size_t *,
+        size_t *, size_t *, size_t *, size_t *);
+static void newProcess(const char *, char *, const char *, econf_file *);
+static void usage(void);
+static void changeRootDir(char *);
+
+int main (int argc, char *argv[])
+{
+    static const char CONFDIR[] = "/.config";
+    econf_err error;
+    char *suffix = NULL; /* the suffix of the filename e.g. .conf */
+    char *posLastDot;
+    char path[PATH_MAX]; /* the path of the config file */
+    char home[PATH_MAX]; /* the path of the home directory */
+    char filename[PATH_MAX]; /* the filename without the suffix */
+    char filenameSuffix[PATH_MAX]; /* the filename with the suffix */
+    char pathFilename[PATH_MAX]; /* the path concatenated with the filename and the suffix */
+    char rootDir[PATH_MAX] = "/etc";
+    char usrRootDir[PATH_MAX] = "/usr/etc";
+    uid_t uid = getuid();
+    uid_t euid = geteuid();
+
+    memset(path, 0, PATH_MAX);
+    memset(home, 0, PATH_MAX);
+    memset(filename, 0, PATH_MAX);
+    memset(pathFilename, 0, PATH_MAX);
```

Comment:
`Memset()` should initialize the arrays with value `0`, since when creating arrays in C it is unknown which content the array has. This is just a precaution here. Furthermore, I learned that every array and every variable should be initialized after declaration/definition.